tests/rados/test_defunct_ssh_connections.py

```diff
@@ -0,0 +1,203 @@
+"""
+
+import subprocess
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.mgr_workflows import MgrWorkflows
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test to verify Manager Failover and check for Defunct SSH Processes
+    Test Steps:
+        1. Failover of Active Manager Continuously
+        2. Check for Defunct SSH Processes
+        3. Ensure system stability after failover
+        4. Application Workaround (if applicable) for defunct SSH
+        5. Ensure No Crashes During Testing
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    rhbuild = config.get("rhbuild")
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mgr_obj = MgrWorkflows(node=cephadm)
+
+    try:
+        # Step 1.1: Ensure the cluster is healthy
+        log.info("Checking cluster health...")
+        rados_obj.log_cluster_health()
+
+        log.info("Step 1.2: Identify the active Ceph manager")
+        # Using the get_active_mgr method to get the active manager
+        active_mgr = mgr_obj.get_active_mgr()  # Get active manager using your method
+        log.info("Current active manager is: {}".format(active_mgr))
+
+        # Step 1.3: Initiate manager failover by restarting the current active manager
+        log.info("Step 1.3: Initiating manager failover")
+        mgr_obj.set_mgr_fail(
+            host=active_mgr
+        )  # Use the method to fail the active manager
+
+        # Step 1.4: Verify failover success
+        log.info("Step 1.4: Verifying the failover")
+        mgr_stats_after_failover = (
+            mgr_obj.get_mgr_stats()
+        )  # Get the mgr stats after failover
+        new_active_mgr = mgr_stats_after_failover[
+            "active_name"
+        ]  # Extract new active manager from stats
+        log.info("New active manager is: {}".format(new_active_mgr))
+
+        if active_mgr == new_active_mgr:
+            log.error("Failover did not occur as expected")
+            return 1
+
+        # Repeat Steps 1.3 and 1.4 multiple times to simulate continuous failovers
+        failover_cycles = 5
+        for i in range(failover_cycles):
+            log.info(
+                "Cycle {}/{}: Initiating manager failover".format(
+                    i + 1, failover_cycles
+                )
+            )
+            mgr_obj.set_mgr_fail(host=active_mgr)  # Fail the current active manager
+            time.sleep(10)  # Ensure enough time for failover
+
+            log.info(
+                "Cycle {}/{}: Verifying the failover".format(i + 1, failover_cycles)
+            )
+            mgr_stats_after_failover = (
+                mgr_obj.get_mgr_stats()
+            )  # Get the mgr stats after failover
+            new_active_mgr = mgr_stats_after_failover["active_name"]
```

Comment:
If mgr stats are not being logged, why not just fetch the new mgr using the method used earlier - mgr_obj.get_active_mgr()